Backend/Core/Helpers/ExtensionMethods.cs

```diff
@@ -88,7 +88,7 @@ public static void OrderNullCheck(Order order)
 
         public static void OrderNotConfirmedCheck(Order order)
         {
-            if (order == null)
+            if (!order.IsAcceptedByClient || !order.IsAcceptedByCourier)
```

Comment:
No. You aren't using this specification to check whether the order is confirmed by both the customer and the courier